actions/upload_sessions.go

```diff
@@ -82,6 +95,47 @@ func (usr *UploadSessionResource) Create(c buffalo.Context) error {
 	return c.Render(200, r.JSON(res))
 }
 
+// Update uploads a chunk associated with an upload session.
+func (usr *UploadSessionResource) Update(c buffalo.Context) error {
+	req := uploadSessionUpdateReq{}
+	parseReqBody(c.Request(), &req)
+
+	// Get session
+	tx := c.Value("tx").(*pop.Connection)
+	uploadSession := &models.UploadSession{}
+	err := tx.Find(uploadSession, c.Param("id"))
+	if err != nil || uploadSession == nil {
+		c.Render(400, r.JSON(map[string]string{"Error finding session": errors.WithStack(err).Error()}))
+		return err
+	}
+
+	// Update dMaps to have chunks async
+	go func() {
+		// Map over chunks from request
+		// TODO: Batch processing DB upserts.
+		dMaps := make([]models.DataMap, len(req.chunks))
+		for i, chunk := range req.chunks {
+			// Fetch DataMap
+			dm := models.DataMap{}
+			tx.RawQuery(
+				"SELECT * from data_maps WHERE genesis_hash ? AND chunk_idx = ?", uploadSession.GenesisHash, chunk.idx).First(&dm)
+
+			// Updates dmap in DB.
+			dm.Message = chunk.data
+			dm.Status = models.Unassigned
+			tx.ValidateAndSave(&dm)
+
+			dMaps[i] = dm
+		}
+
+		// Should we still do this here?
```

Comment:
@rfornea Do we still want to process chunks here, or just let the cron job take care of this?

Reply:
We can pick it up in the cron job.  We can change the frequency of the cron task to run as often as we like.